process/block/blockProcessor.go

```diff
@@ -105,6 +107,29 @@ func (bp *blockProcessor) PrepareBlockForDB(
 	return elasticBlock, nil
 }
 
+func getTxsCount(header coreData.HeaderHandler) (numTxs, notarizedTxs uint32) {
+	numTxs = header.GetTxCount()
+
+	if core.MetachainShardId != header.GetShardID() {
+		return numTxs, notarizedTxs
+	}
+
+	metaHeader, ok := header.(*nodeBlock.MetaBlock)
+	if !ok {
+		return 0, 0
+	}
+
+	notarizedTxs = metaHeader.TxCount
+	numTxs = 0
+	for _, mb := range metaHeader.MiniBlockHeaders {
```

Comment:
although this works because we do not use the `mb` pointer directly, the iteration is done over a slice of values